File: tradex-middleware/src/controller/ReportController.js
// controllers/ReportController.js
const express = require('express');
const router = express.Router();
const reportService = require('../service/ReportService');

console.log('ReportController loaded');

// ----------------- Get portfolio report -----------------
router.get('/portfolio/:clientId', async (req, res) => {
    console.log('GET /portfolio/:clientId', req.params.clientId);
    try {
        const report = await reportService.getPortfolioReport(req.params.clientId);
        res.json(report);
    } catch (err) {
        console.error('Error in /portfolio/:clientId:', err);
        res.status(500).json({ error: err.message });
    }
});

// ----------------- Get trade history report -----------------
router.get('/trade-history/:clientId', async (req, res) => {
    console.log('GET /trade-history/:clientId', req.params.clientId);
    try {
        const report = await reportService.getTradeHistoryReport(req.params.clientId);
        res.json(report);
    } catch (err) {
        console.error('Error in /trade-history/:clientId:', err);
        res.status(500).json({ error: err.message });
    }
});

module.exports = router;

File: tradex-service/src/main/java/com/example/tradex_service/Controller/ReportController.java
package com.example.tradex_service.Controller;

import com.example.tradex_service.DTO.Response.PortfolioReportDto;
import com.example.tradex_service.DTO.Response.TradeHistoryReportDto;
import com.example.tradex_service.Service.ReportService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@CrossOrigin(origins = "*")
@RestController
@RequestMapping("/reports")
public class ReportController {

    @Autowired
    private ReportService reportService;

    @GetMapping("/portfolio/{clientId}")
    public ResponseEntity<PortfolioReportDto> getPortfolioReport(@PathVariable String clientId) {
        PortfolioReportDto report = reportService.getPortfolioReport(clientId);
        return ResponseEntity.ok(report);
    }

    @GetMapping("/trade-history/{clientId}")
    public ResponseEntity<TradeHistoryReportDto> getTradeHistoryReport(@PathVariable String clientId) {
        TradeHistoryReportDto report = reportService.getTradeHistoryReport(clientId);
        return ResponseEntity.ok(report);
    }
}